unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarNamesHUD/AvatarNamesHUDView.cs

```diff
@@ -0,0 +1,119 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+namespace AvatarNamesHUD
+{
+    public interface IAvatarNamesHUDView
+    {
+        void Initialize(int maxAvatarNames);
+        void SetVisibility(bool visibility);
+        void TrackPlayer(PlayerStatus playerStatus);
+        void UntrackPlayer(string userId);
+        void Dispose();
+    }
+
+    public class AvatarNamesHUDView : MonoBehaviour, IAvatarNamesHUDView
+    {
+        [SerializeField] internal RectTransform canvasRect;
+        [SerializeField] internal RectTransform backgroundsContainer;
+        [SerializeField] internal RectTransform namesContainer;
+        [SerializeField] internal RectTransform voiceChatContainer;
+        [SerializeField] internal GameObject backgroundPrefab;
+        [SerializeField] internal GameObject namePrefab;
+        [SerializeField] internal GameObject voiceChatPrefab;
+
+        internal readonly Dictionary<string, AvatarNamesTracker> trackers = new Dictionary<string, AvatarNamesTracker>();
+        internal readonly Queue<AvatarNamesTracker> reserveTrackers = new Queue<AvatarNamesTracker>();
+
+        private bool isDestroyed = false;
+
+        public static IAvatarNamesHUDView CreateView()
+        {
+            AvatarNamesHUDView view = Instantiate(Resources.Load<GameObject>("AvatarNamesHUD")).GetComponent<AvatarNamesHUDView>();
+            view.gameObject.name = "_AvatarNamesHUD";
+            return view;
+        }
+
+        private void OnEnable()
+        {
+            canvasRect.GetComponent<Canvas>().sortingOrder = -2;
```

Comment:
We should move this to somewhere common to the project like `Settings`, So in the future, if we need to change the sorting order in other parts of the project we have some visibility about the current canvas ordering. What do you think?

Reply:
Not sure about that, this is an specific sorting set for the AvatarNames. I dont think any other part of the project should be aware of what sorting order does the AvatarNames canvas have.

Reply:
As you wish!